File: leet_3158_double_xor.go
func duplicateNumbersXOR(nums []int) int {

    res := 0
    S := map[int]bool{}
    for _, n := range nums {
        if S[n] {
            res ^= n
        } else {
            S[n] = true
        }
    }
    return res
}

File: leet_3158_double_xor.cc
class Solution {
public:
    int duplicateNumbersXOR(vector<int>& nums)
    {
        int res = 0;
        set<int> S;
        for (int n: nums)
        {
            if (S.find(n) != end(S))
                res ^= n;
            else
                S.insert(n);
        }
        return res;
    }
};
